test/common/http/http1/codec_impl_test.cc

```diff
@@ -142,6 +142,14 @@ TEST_F(Http1ServerConnectionImplTest, Http11AbsolutePath2) {
   expectHeadersTest(Protocol::Http11, true, buffer, expected_headers);
 }
 
+TEST_F(Http1ServerConnectionImplTest, Http11AbsolutePathWithPort) {
+  TestHeaderMapImpl expected_headers{
+      {":authority", "www.somewhere.com:4532"}, {":path", "/foo/bar"}, {":method", "GET"}};
+  Buffer::OwnedImpl buffer(
+      "GET http://www.somewhere.com:4532/foo/bar HTTP/1.1\r\nHost: bah\r\n\r\n");
+  expectHeadersTest(Protocol::Http11, true, buffer, expected_headers);
+}
```

Comment:
One early suggestion, can you add an integration test with a  port as well?  I assume we'd want route config with host "www.foo.com" to match "www.foo.com:80" and I'm not confident this would work end-to-end

I think you can just tweak the config you added in your prior patch to not have a * domain and then add or copy-and-extend your current absolute-uri test to make sure we match correctly.